packages/vite/src/node/plugins/css.ts

```diff
@@ -720,22 +748,24 @@ export function cssPostPlugin(config: ResolvedConfig): Plugin {
 
       if (chunkCSS) {
         if (config.build.cssCodeSplit) {
+          const cssFullAssetName = ensureFileExt(chunk.name, '.css')
+          // if facadeModuleId doesn't exist or doesn't have a CSS extension,
+          // that means a JS entry file imports a CSS file.
+          // in this case, only use the filename for the CSS chunk name like JS chunks.
+          const cssAssetName =
+            chunk.isEntry &&
+            (!chunk.facadeModuleId || !isCSSRequest(chunk.facadeModuleId))
+              ? path.basename(cssFullAssetName)
+              : cssFullAssetName
+
```

Comment:
I hoisted this part up because at line 811, we can pass the `cssAssetName` as the filename.